### PR TITLE
netCDFConfig: find HDF5 if needed

### PR DESCRIPTION
**Pull request by mathstuf**
See: #1733

### CODE REVIEW

**Comment by WardF** [comment]
I have run into issues with disparities between HDF5's cmake integration implementation between Linux/MacOS and Windows in the past. Let me double-check this by hand but looks good!

**Comment by mathstuf** [comment]
Agreed. I've run into the issue where NetCDF's references to `hdf5_hl-shared` targets end up with failure to find a `hdf5_hl-shared` library on-disk because the target name is never resolved. If there's some way to detect that NetCDF used imported targets from HDF5, then that can be used as the conditional instead.

File: netCDFConfig.cmake.in [hunk 1]
```diff
@@ -40,3 +40,7 @@ set(netCDF_HAS_DAP4 @HAS_DAP4@)
 set(netCDF_HAS_DISKLESS @HAS_DISKLESS@)
 set(netCDF_HAS_MMAP @HAS_MMAP@)
 set(netCDF_HAS_JNA @HAS_JNA@)
+
+if (netCDF_HAS_HDF4 OR netCDF_HAS_HDF5)
+  find_dependency(HDF5)
+endif ()
```
Comment (kmorel) [inline netCDFConfig.cmake.in:46]: To use the `find_dependency` macro, you need to include `CMakeFindDependencyMacro`.

```suggestion
include(CMakeFindDependencyMacro)
if (netCDF_HAS_HDF4 OR netCDF_HAS_HDF5)
  find_dependency(HDF5)
endif ()
```

Comment (mathstuf) [reply netCDFConfig.cmake.in]: Whoops, yep. Thanks. I added it inside the `if` block though.

Comment (kmorel) [reply netCDFConfig.cmake.in]: That's fine with me.

Can you update this in the ParaView superbuild? That's where I ran into this problem.

**Review by WardF — APPROVED** [review]

**Review by WardF — APPROVED** [review]

**Pull request merged**